## 퀀트 전략을 이용한 종목선정

퀀트 투자는 크게 포트폴리오 운용 전략과 트레이딩 전략으로 나눌 수 있다. 포트폴리오 운용 전략은 과거 주식 시장을 분석해 좋은 주식의 기준을 찾아낸 후 해당 기준에 만족하는 종목을 매수하거나, 이와 반대에 있는 나쁜 주식을 공매도하기도 한다.  
반면 트레이딩 전략은 주식이 오르거나 내리는 움직임을 연구한 후 각종 지표를 이용해 매수 혹은 매도하는 전략이다.

> 공매도란?  
공매도란 주식을 보유하고 있지 않은 상태에서 주식을 먼저 팔고, 나중에 주식을 사서 갚는 방식이다. 주가가 하락할 것이라고 예상하여 증권사로부터 주식을 빌려 현재 가격에 팔고 나중에 주가가 떨어졌을 때 더 낮은 가격에 사서 갚아 차익을 남길 수 있는 방법이다.

## 팩터 이해하기

주식의 수익률에 영향을 미치는 특성들을 '팩터(Factor)'라고 한다.

## 베타 이해하기

전체 주식시장의 움직임은 개별 주식의 수익률에 가장 크게 영향을 주는 팩터일 수 밖에 없다. 아무리 좋은 주식도 주식시장이 폭락한다면 같이 떨어지며, 아무리 나쁜 주식도 주식시장이 상승한다면 대부분 같이 오르기 마련이다.

개별 주식이 전체 주식시장의 변동에 반응하는 정도를 나타낸 값이 베타다. 베타가 1이라는 뜻은 주식시장과 움직임이 정확히 같다는 뜻으로서 시장 그 자체를 나타낸다. 베타가 1.5라는 뜻은 주식시장이 수익률이 +1%일 때 개별 주식의 수익률은 +1.5% 움직이며, 반대로 주식시장의 수익률이 -1%일 때 개별 주식의 수익률은 -1.5% 움직인다는 뜻이다. 반면 베타가 0.5라면 주식시장 수익률의 절반 정도만 움직인다.

일반적으로 상승장이 기대될 때는 베타가 큰 주식에, 하락장이 기대될 때는 베타가 낮은 주식에 투자하는 것이 좋다.

#### 자산가격결정모형(CAMP: Capital Asset Pricing Model)

$$ R_i = R_f + \beta_i \times [R_m - R_f] $$

| 구분 | 회귀분석모형 | 자산가격결정모형 | 
| --- | --- | --- |
| 상수항 | $a$ | $R_f$ (무위험수익률) |
| 독립변수 | $x$ | $R_m - R_f$ (시장위험 프리미엄) |
| 종속변수 | $y$ | $R_i$ (개별주식의 수익률) |
| 회귀계수 | $b$ | $\beta_i$ (개별주식의 베타) |베타) |

> 무위험수익률: 국채 같은 위험이 거의 없는 자산에서 기대할 수 있는 수익률  
> 시장위험 프리미엄: 전체 시장 수익률에서 무위험수익률을 뺀 값

통계학에서 회귀계수는 $\beta = \frac{cov(x, y)}{\sigma_x^2}$ 형태로 구할 수 있으며, $x$와 $y$에 각각 시장수익률과 개별주식의 수익률을 대입할 경우 개별주식의 베타는 $\beta_i = p(i, m) \times \frac{\sigma_i}{\sigma_m}$ 형태로 구할 수 있다.

>$cov(x, y)$: 시장수익률과 개별주식의 수익률의 공분산  
$\sigma_x^2$: 시장수익률의 분산  
$p(i, m)$: 시장수익률과 개별주식의 수익률 간 상관계수  
$\sigma_i$: 개별주식 수익률의 표준편차  
$\sigma_m$: 시장수익률의 표준편차

### 베타 계산하기

베타를 구하는 방법을 알아보기 위해 주식시장에 대한 대응치로 KOSPI 지수를, 개별주식으로는 전통적 고베타주인 증권주를 이용한다.

1. KOSPI 코드(^KS11)와 증권주인 키음증권의 티커(039490.KS)를 입력한다.

In [1]:
import yfinance as yf
import pandas as pd

tickers = ['^KS11', '039490.KS']

2. download() 함수를 이용하여 해당 티커들의 2016년부터 2021년까지 데이터를 다운로드 받는다.

In [2]:
all_data = {}
for ticker in tickers:
    all_data[ticker] = yf.download(ticker,
                                  start='2016-01-01',
                                  end='2021-12-31')

YF.download() has changed argument auto_adjust default to True


[*********************100%***********************]  1 of 1 completed
[*********************100%***********************]  1 of 1 completed


3. 종가(Close)에 해당하는 열만 선택한 후, 데이터프레임 형태로 만들어 준다.

In [3]:
prices = pd.DataFrame({
    tic: data['Close'][tic] for tic, data in all_data.items()
})

4. pct_change() 함수를 통해 수익률을 계산한 후, NA 데이터는 삭제한다.

In [4]:
ret = prices.pct_change().dropna()
ret

^KS11  039490.KS
Date                           
2016-01-05  0.006134   0.028428
2016-01-06 -0.002642  -0.006504
2016-01-07 -0.010959  -0.026187
2016-01-08  0.006979   0.003361
2016-01-11 -0.011879  -0.028476
...              ...        ...
2021-12-24  0.004756   0.000000
2021-12-27 -0.004276  -0.009132
2021-12-28  0.006898   0.004608
2021-12-29 -0.008923   0.023697
2021-12-30 -0.005225  -0.009259

[1472 rows x 2 columns]

증권주를 대상으로 베타를 구하기 위한 회귀분석을 실시한다. 자산가격결정모형의 수식인 $R_i = R_f + \beta_i \times [R_m - R_f]$ 에서 편의를 위해 무위험 수익률인 $R_f$를 0으로 가정하면, $R_i = \beta_i \times R_m$의 형태로 나타낼 수 있다. 이 중 $R_m$는 독립변수인 주식시장의 수익률을 의미하고, $R_i$는 종속변수인 개별 주식의 수익률을 의미한다.

1. 알파를 계산하기 위해 intercept(절편)에 해당하는 열에 1을 입력한다.

In [6]:
import statsmodels.api as sm

ret['intercept'] = 1

2. 종속변수에는 증권주, 독립변수에는 KOSPI 지수 수익률 및 절편을 입력한다.
3. statsmodels 패키지의 OLS() 함수를 통해 손쉽게 선형회구분석을 실시할 수 있으며, 회귀분석의 결과를 reg 변수에 저장한다.

In [7]:
reg = sm.OLS(ret[['039490.KS']], ret[['^KS11', 'intercept']]).fit()

In [8]:
reg.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:              039490.KS   R-squared:                       0.366
Model:                            OLS   Adj. R-squared:                  0.366
Method:                 Least Squares   F-statistic:                     849.4
Date:                Tue, 08 Apr 2025   Prob (F-statistic):          9.05e-148
Time:                        15:39:53   Log-Likelihood:                 3683.2
No. Observations:                1472   AIC:                            -7362.
Df Residuals:                    1470   BIC:                            -7352.
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
^KS11          1.4321      0.049     29.145      0.000       1.336       1.528
intercept      0.0003      0.001      0.550      0.582      -0.001       0.001
==============================================================================
Omnibus:                      221.231   Durbin-Watson:                   2.218
Prob(Omnibus):                  0.000   Jarque-Bera (JB):              568.697
Skew:                           0.812   Prob(JB):                    3.23e-124
Kurtosis:                       5.576   Cond. No.                         95.1
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

In [9]:
print(reg.params)

^KS11        1.432088
intercept    0.000285
dtype: float64


summary() 메서드는 회구분석 결과의 요약 정보를 보여준다. 또한 reg.params을 통해 베타(^KS11) 및 알파(intercept)에 해당하는 값만 선택할 수도 있다.

회귀분석의 결과 테이블 중 베타를 나타내는 부분은 coef 이다. 베타값이 약 1.43으로 높아 증권주의 특성인 고베타주임이 확인되며, t값 또한 약 29로써 유의성을 가르는 2 보다 크므로 결과가 유의하다고 볼 수 있다. 반면 알파(intercept)는 0.0002 수준이며 t값 역시 0.376으로 매우 낮아, 증권주의 수익률은 주식 시장에 대한 노출도인 베타를 제외하고 나면 초과 수익이 없다고 볼 수 있다.